open-craft/opencraft
# [BB-986] Provide users with sample privacy policy

Adds a field to the registration form to allow users to provide a URL to a Privacy Policy when signing up. The Privacy Policy URL is optional - a default can be specified with the `DEFAULT_PRIVACY_POLICY_URL` Django setting or environment variable. Existing instances without a Privacy Policy already set may be edited without needing to provide a Privacy Policy URL. When the Privacy Policy URL is provided, the `EDXAPP_LMS_ENV_EXTRA` has `MKTG_URL_OVERRIDES` added and populated with the provided value in the `PRIVACY` element.

## Testing Instructions

* Merge this branch and apply migrations.
* Start the dev server, specifying a valid URL in the `DEFAULT_PRIVACY_POLICY_URL` environment variable: `DEFAULT_PRIVACY_POLICY_URL=http://localhost/privacy make run.dev`
* Observe the `Privacy Policy URL` field in the registration form.
* Attempt to submit the application with an invalid URL in the field.
  - The registration attempt should be rejected.
* Provide a valid URL for the field and submit the application.
  - The registration attempt should succeed.
* Provision an appserver from the newly-created instance.
  - Observe in the appserver's `Combined ansible vars` that:
    + `EDXAPP_LMS_ENV_EXTRA` contains `MKTG_URL_OVERRIDES`, which has your provided URL as the `PRIVACY` element.
